### PR TITLE
Add RCL auto acknowledge for CTP 24E

### PR DESCRIPTION
**Pull request by lieselwd**
The current workflow on the initial state of this PR

- Pilot submits RCL within time constraint boundaries. System responds with "RCL/CLA RECEIVED\nCONTINUE TO MONITOR NATTRAK UNTIL ENTERING OCA"
- RCL appears in controller "Pending" list. When actioning the clearance, the controller can see it was already acknowledged by the system.
- The controller can do conflict checks and other matters. If required, they can issue an override re-clearance. Otherwise, they can move the RCL to the Processed List. 
- If the RCL is moved to the Processed list, a CLX is created and the pilot receives the message:
 
image

Please provide feedback on this model and let me know what adjustments need to be made via reply in this issue.

### CODE REVIEW

File: app/Http/Livewire/Controllers/LtNewPending.php [hunk 1]
```diff
@@ -5,14 +5,17 @@
 use App\Http\Controllers\ClxMessagesController;
 use App\Models\Track;
 use Illuminate\Database\Eloquent\Builder;
+use phpDocumentor\Reflection\Types\Boolean;
```
Comment (williammck) [inline app/Http/Livewire/Controllers/LtNewPending.php:8]: where'd this guy come from

File: app/Models/RclMessage.php [hunk 2]
```diff
@@ -171,6 +175,24 @@ public function scopeCleared(Builder $query): Builder
         return $query->whereNotNull('clx_message_id');
     }
 
+    /**
+     * @param Builder $query
+     * @return Builder
+     */
+    public function scopeNotAcknowledged(Builder $query): Builder
+    {
+        return $query->whereNot('is_acknowledged');
```
Comment (williammck) [inline app/Models/RclMessage.php:184]: needs to be
```suggestion
        return $query->where('is_acknowledged', false);
```

File: app/Models/RclMessage.php [hunk 3]
```diff
@@ -171,6 +175,24 @@ public function scopeCleared(Builder $query): Builder
...
+    }
+
+    /**
+     * @param Builder $query
+     * @return Builder
+     */
+    public function scopeAcknowledged(Builder $query): Builder
+    {
+        return $query->where('is_acknowledged');
```
Comment (williammck) [inline app/Models/RclMessage.php:193]: needs to be
```suggestion
        return $query->where('is_acknowledged', true);
```

File: app/Enums/RclResponsesEnum.php [hunk 4]
```diff
@@ -11,7 +11,8 @@ enum RclResponsesEnum: string
     case Cancelled = "RCL/CLA REJECTED\nCLEARANCE CALLED REVERT TO VOICE PROCEDURES";
     case Invalid = "RCL REJECTED\nINVAILD %s\nRESUBMIT YOUR REQUEST";
     case Contact = "RCL/CLA RECEIVED\nCONTACT %s BY VOICE";
-    case Acknowledge = "RCL/CLA RECEIVED\nFLY CURRENT FLIGHT PLAN OR AS AMENDED BY ATC\nCONTINUE TO MONITOR NATTRAK UNTIL ENTERING OCA";
+    case Acknowledge = "RCL/CLA RECEIVED\nCONTINUE TO MONITOR NATTRAK UNTIL ENTERING OCA";
+    case AcknowledgeMoved = "RCL/CLA RECEIVED BY %s\nCONTINUE TO MONITOR NATTRAK";
```
Comment (williammck) [inline app/Enums/RclResponsesEnum.php:15]: defer to others on whether this is the correct verbiage

File: app/Http/Requests/RclMessageRequest.php [hunk 5]
```diff
@@ -88,9 +88,9 @@ public function withValidator($validator)
                 /** Entry fix time requirement */
                 if (config('app.rcl_time_constraints_enabled') && strlen($this->entry_time) == 4) {
                     if (!$this->entryTimeWithinRange($this->entry_time)) {
-//                        if (config('app.rcl_auto_acknowledgement_enabled')) {
-//                            $this->cpdlcService->sendMessage(author: DatalinkAuthorities::SYS, recipient: $this->callsign, recipientAccount: Auth::user(), message: sprintf(RclResponsesEnum::Contact->value, strtoupper(DatalinkAuthorities::OCEN->description())), caption: RclResponsesEnum::Contact->text());
-//                        }
+                        if (config('app.rcl_auto_acknowledgement_enabled')) {
+                            $this->cpdlcService->sendMessage(author: DatalinkAuthorities::SYS, recipient: $this->callsign, recipientAccount: Auth::user(), message: sprintf(RclResponsesEnum::Contact->value, strtoupper(DatalinkAuthorities::OCEN->description())), caption: RclResponsesEnum::Contact->text());
```
Comment (williammck) [inline app/Http/Requests/RclMessageRequest.php:92]: i'm still confused why this is here. isn't this the validator method? do we want to send this message if things don't validate?

Comment (lieselwd) [reply app/Http/Requests/RclMessageRequest.php:92]: That's something I was also thinking about. It seems the IRL implementation is to send that message back, which I'd be ok with just doing, but also VATSIM pilots if not stopped forcefully will send bogus requests that will have to be rejected anyway. The message there for 'realism', the validation there to stop bogus requests.

File: app/Http/Livewire/Controllers/LtNewPending.php (hunk 1 above)
Comment (lieselwd) [reply app/Http/Livewire/Controllers/LtNewPending.php:8]: not there on my local copy, who knows. Will be removed either way.

File: app/Enums/RclResponsesEnum.php (hunk 4 above)
Comment (lieselwd) [reply app/Enums/RclResponsesEnum.php]: https://github.com/vatsimnetwork/nattrak/pull/142/commits/b0d865a4c8147748458d9e79d73580e2bdd75e44

**Review by williammck — APPROVED** [review]

**Pull request merged**